ZenPacks/zenoss/Import4/migration.py

```diff
@@ -76,16 +112,96 @@ def wipe(self):
 
     def doImport(self):
         self.__NOT_YET__()
-        logging.warning("-- To be overriden!")
+        self.log.warning("-- To be overriden!")
         return
 
     def reportProgress(self, progress):
         # callback to self.progress if register at creation
-        logging.debug(progress)
+        progress = progress.rstrip("\n\r\t ").lstrip("\n\r")
+        self.log.debug(progress)
         if self.progress:
-            self.progress(progress)
+            self.progress(progress + "\n")
 
     def postvalidate(self):
         self.__NOT_YET__()
-        logging.warning("-- To be overriden!")
+        self.log.warning("-- To be overriden!")
+        return
+
+    def startZenoss(self):
+        # now bring back zenoss processes AFTER the new zope image is committed
+        self.reportProgress('Restarting zenoss services...')
+        _cmd = '%s/imp4util.py --log-level=%s start_all_svcs' % (self.binpath, self.args.log_level)
+        self.log.debug('-> %s' % _cmd)
+        self.exec_cmd(_cmd)
+        return
+
+    '''
+    Utility methods
+    '''
+    # execute a command and pocesses the stdout/stderr
+    def exec_cmd(self, cmd):
+        # prep for the error log file for the restoration command
+        _errfile = tempfile.NamedTemporaryFile(
+            mode='w+b', dir=self.tempDir, prefix='_zen', delete=False)
+
+        self.log.debug('Executing %s ...' % cmd)
+
+        proc = subprocess.Popen(cmd, shell=True,
```

Comment:
The ```call()``` and ```check_*()``` functions are more commonly used out of the ```subprocess``` module, why not use those?